### PR TITLE
Attempt to add explorer and ows indexing

### PR DESCRIPTION
**Pull request by alexgleith**
OWS indexing requires an init container to load config, which is documented here: https://airflow.readthedocs.io/en/latest/howto/operator/kubernetes.html#

The explorer process should be straightforwards.

### CODE REVIEW

File: dags/sentinel-2_indexing.py [hunk 1]
```diff
@@ -42,8 +46,31 @@
...
+    sub_path=None,
+    read_only=True
+)]
+
+init_container = k8s.V1Container(
+  name="init-container",
+  image="geoscienceaustralia/deafrica-config:0.1.1-unstable.305.ge30a170",
+  volume_mounts=init_container_volume_mounts,
+  command=["bash", "-cx"],
+  args=["echo 10"]
```
Comment (tom-butler) [inline dags/sentinel-2_indexing.py:72]:
```suggestion
  args=["cp", "-f", "/opt/dea-config/services/ows_cfg.py",  "/env/config/ows_cfg.py"]
```

File: dags/sentinel-2_indexing.py [hunk 2]
```diff
@@ -42,8 +46,31 @@
...
+    name='ows-config',
+    sub_path=None,
+    read_only=True
+)]
+
+init_container = k8s.V1Container(
+  name="init-container",
+  image="geoscienceaustralia/deafrica-config:0.1.1-unstable.305.ge30a170",
+  volume_mounts=init_container_volume_mounts,
+  command=["bash", "-cx"],
```
Comment (tom-butler) [inline dags/sentinel-2_indexing.py:71]: don't need this, the default image command is fine

File: dags/sentinel-2_indexing.py [hunk 3]
```diff
@@ -68,30 +94,38 @@
...
         task_id="indexing-task",
         get_logs=True,
+        is_delete_operator_pod=True,
     )
 
-    # TODO: implement
     UPDATE_RANGES = KubernetesPodOperator(
         namespace="processing",
         image=OWS_IMAGE,
         cmds=["datacube-ows-update"],
```
Comment (tom-butler) [inline dags/sentinel-2_indexing.py:103]: need to run the --views first, it's a 2 step operation to update ranges now

1 `datacube-ows-update --views --blocking` updates the sql views
2 `datacube-ows-update s2_l2a` updates the product

File: dags/sentinel-2_indexing.py [hunk 4]
```diff
@@ -68,30 +94,38 @@
...
-        arguments=["--help"],
+        arguments=["s2_l2a"],
         labels={"step": "ows"},
         name="ows-update-ranges",
         task_id="update-ranges-task",
         get_logs=True,
+        volumes=[volume],
+        init_containers=[init_container],
+        is_delete_operator_pod=True,
```
Comment (tom-butler) [inline dags/sentinel-2_indexing.py:111]: need to set the required env vars too:
```
      # Defaults are defined in .env file
      DB_HOSTNAME: ${DB_HOSTNAME}
      DB_PORT: ${DB_PORT}
      DB_USERNAME: ${DB_USERNAME}
      DB_PASSWORD: ${DB_PASSWORD}
      DB_DATABASE: ${DB_DATABASE}
      # Where the config file is (must match the volume mount)
      WMS_CONFIG_PATH: /env/config/ows_cfg.py
      # Path from the PYTHONPATH to the config object (default PYTHONPATH is /env) 
      DATACUBE_OWS_CFG: config.ows_cfg.ows_cfg
      AWS_DEFAULT_REGION: ${AWS_REGION}
      # Talk to AWS without using credentials
      AWS_NO_SIGN_REQUEST: "${AWS_NO_SIGN_REQUEST}"
```

Comment (alexgleith) [reply dags/sentinel-2_indexing.py:111]: The `DEFAULT_ARGS` get passed in to all of them. The AWS-specific flags do need to be set, though.

File: dags/sentinel-2_indexing.py [hunk 5]
```diff
@@ -42,8 +46,31 @@
 }
 
 INDEXER_IMAGE = "opendatacube/datacube-index:0.0.7"
-OWS_IMAGE = "opendatacube/ows:0.14.1"
-EXPLORER_IMAGE = "opendatacube/dashboard:2.1.6"
+OWS_IMAGE = "opendatacube/ows:0.14.6"
```
[the remark this answers is not in the document]
Comment (alexgleith) [reply dags/sentinel-2_indexing.py:49]: I copied this from the deployed DE Africa version, from here: https://bitbucket.org/geoscienceaustralia/datakube-apps/src/74b0ae4f3604d57669f7abcbf6164b19408abc85/workspaces/deafrica-prod/web/04_ows.yaml#lines-27

**Pull request merged**